### ISSUE TITLE
Comparison between sealed traits implementation with ignored fields is not working

### ISSUE THREAD

**Issue by rlopezlozano** [softwaremill/diffx#41]
Here an example:

```scala
import com.softwaremill.diffx.scalatest.DiffMatcher
import com.softwaremill.diffx.{Derived, Diff}
import org.scalatest.{FlatSpec, Matchers}

class Test extends FlatSpec with Matchers with DiffMatcher {

  sealed trait Base {
    def id: Int
    def name: String
  }

  final case class SubtypeOne(id: Int, name: String) extends Base

  final case class SubtypeTwo(id: Int, name: String) extends Base

  case class Example(base: Base)

  case class Example1(bases: List[Base])

  implicit val b: Diff[Base] = Derived[Diff[Base]].ignore(_.id)
  implicit val e: Diff[Example] = Derived[Diff[Example]].ignore(_.base.id)
  implicit val ex: Diff[Example1] = Derived[Diff[Example1]].ignore(_.bases.each.id)

  val baseRight = SubtypeOne(1, "one")
  val baseRight1 = SubtypeOne(1, "one")
  val baseLeft = SubtypeOne(2, "one")

  val exampleRight = Example(baseRight)
  val exampleLeft = Example(baseLeft)

  val examplesRight = Example1(List(baseRight))
  val examplesLeft = Example1(List(baseLeft))

  baseLeft should matchTo(baseRight)
  exampleLeft should matchTo(exampleRight)
  examplesRight should matchTo(examplesLeft)
}
```

**Comment by ghostbuster91** [comment]
Thanks for reporting! I will release patched version soon.

**Comment by ghostbuster91** [comment]
Released in 0.3.9, keep in mind that in order `  baseLeft should matchTo(baseRight)` to work you have to explicitly upcast them to `Base` before comparing, although I will think how that could be simplified.

**Comment by rlopezlozano** [comment]
That's great! Thanks! :)
